### ISSUE TITLE
[java] SimplifiedTernary: no violation for 'condition ? true : false' case

### ISSUE THREAD

**Issue by dykov** [pmd/pmd#3603]
**Affects PMD Version:** 6.40.0

**Rule:** [SimplifiedTernary](https://pmd.github.io/latest/pmd_rules_java_design.html#simplifiedternary)

**Description:** There is no violation if both of ternary returns are boolean.

**Code Sample demonstrating the issue:**

```java
return condition ? true : false;
```

**Expected outcome:**

PMD should report a violation at line 1, but doesn't. This is a false-negative.

**Running PMD through:** *Maven | PMD Rule Designer*

**Comment by dykov** [comment]
Could the [rule xpath](https://pmd.github.io/latest/pmd_rules_java_design.html#simplifiedternary) be simplified as follows?
```
//ConditionalExpression
[
    (Expression/PrimaryExpression/*/Literal/BooleanLiteral) 
    or 
    (PrimaryExpression/*/Literal/BooleanLiteral)
]
```
Or there are any false-positive cases?

**Comment by VoidxHoshi** [comment]
I'm new here, and I want to contribute to this. Could the maintainers please assign this to me? 
thx
😄 
I will try to fix it.
